lib/src/rules/always_require_non_null_named_parameters.dart

```diff
@@ -50,24 +51,26 @@ class AlwaysRequireNonNullNamedParameters extends LintRule
   @override
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
-    final visitor = _Visitor(this);
+    final visitor = _Visitor(this,
+        isNullSafety: context
+            .currentUnit.unit.declaredElement.library.isNonNullableByDefault);
```

Comment:
We've actually been using a slightly different pattern. First, we added `isEnabled` to `LinterContext` to make it a little shorter. Second, we're using a pattern like
```
if (is not non-nullable) {
  registry.add...
}
```
By not registering in cases where the rule shouldn't run, the visitor methods won't even be called.

Reply:
Ah, excellent. Done.